insights/core/__init__.py

```diff
@@ -365,7 +365,10 @@ class YAMLParser(Parser, LegacyItemAccess):
     A parser class that reads YAML files.  Base your own parser on this.
     """
     def parse_content(self, content):
-        self.data = yaml.safe_load('\n'.join(content))
+        if type(content) is list:
+            self.data = yaml.safe_load('\n'.join(content))
+        else:
+            self.data = yaml.safe_load(content)
```

Comment:
@lhuett do you need to add a test for this?

Reply:
@bfahr I added test_branch_info.py that tests the non list situation I added to support branch_info.py. I believe that the only parser class defined in __init__ that actually had a test for it individually is XMLParser. I could add one for this a guess.

Reply:
Would you mind to add a test for this parser?

Reply:
Created a new test for core Parser class YAMLParser(Parser, LegacyItemAccess)
test_yamlparser.py